### PR TITLE
Rename tabs of QENS Fitting interface

### LINKED ISSUE #36853: Rename tabs and tab files in QENS Fitting
**Description:**
We want to rename the tabs in the QENS Fitting interface to provide a better description of what they are doing. From Spencer:

```
With the renaming the tabs can have their names changed with Fit removed as it’s redundant/obvious. Then 
MSD remains just that as does I(Q,t)
ConvFit becomes Convolution - better definition 
F(Q) becomes Function(Q) - again better definition
```

We will also want to update the naming of the respective files and classes.

**Acceptance Criteria**
- The tabs in the QENS Fitting interface have been renamed as described above.
- The files and classes for these tabs have been renamed
- The documentation has been updated.

### PR DESCRIPTION
**Pull request by adriazalvarez**
### Description of work
After renaming the Data Analysis tab to QENS Fitting, we think the names of the tabs are a bit redundant (MSD Fit, I(Q,t) Fit, ConvFit...) as it is clear from the interface name that its purpose is to fit different types of data.  The new names for the tabs are: MSD, I(Q,t), Convolution and Function(Q). In summary: 
- Changed tabs names. 
- Changed related classes names. 
- Updated docs and dev-docs references. 

More information can be found in #36853. 

Additionally, I have added a tooltip on the tabs, which will raise a helpful info message of what each tab does whenever the mouse is hovered on that tab. 

To be noted: The `I(Q,t) Fit` is renamed as `I(Q, t)`  on the GUI, but the underlying classes and files preserve the name starting with `IqtFit...` as there is a nother `Iqt..` class on the Data Processor interface that interferes with it, and it will raise confusion if both classes share the same name (for example there would be two `IqtModel` files). 

Fixes #36853 . 

### To test:
- Build dev-docs. 
- Follow `Inelastic QENS Fitting Manual Testing` instructions on the built docs, making sure the instructions contain the updated tab names. 
- Make sure the manual tests run without problem, and the tabs on `QENS Fitting` interface have the new names. 
- Build docs, check the `Interfaces`->`QENS Fitting` interface page and check everything looks updated with the new names and properly formatted. 

*Added release notes*

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: qt/scientific_interfaces/Inelastic/QENSFitting/FitTabConstants.h [hunk 1]
```diff
@@ -76,8 +76,8 @@ inline auto templateSubTypes() {
 
 } // namespace IqtFit
```
Comment (robertapplin) [inline qt/scientific_interfaces/Inelastic/QENSFitting/FitTabConstants.h:77]: Should this namespace be changed to `Iqt` or similar?

**Review by robertapplin — CHANGES_REQUESTED** [review]
This change has made a big improvement to the code readability, and also will make a big difference from a user-facing perspective. Thanks for the attention to detail on this

I spotted a couple of things listed below. Also, there is a lot of locations where IqtFit remains. Is this in order to avoid confusion with the Iqt tab on the Data Processor interface?

File: qt/scientific_interfaces/Inelastic/QENSFitting/FitTabConstants.h (hunk 1 above)
Comment (adriazalvarez) [reply qt/scientific_interfaces/Inelastic/QENSFitting/FitTabConstants.h:77]: I have just left as IqtFit as the model is similar.

File: qt/scientific_interfaces/Inelastic/QENSFitting/QENSFitting.ui [hunk 2]
```diff
@@ -58,22 +61,34 @@
...
+       </attribute>
+       <attribute name="toolTip">
+        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;MSD: &lt;/span&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;Mean Square Displacements for different fit models.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </attribute>
       </widget>
-      <widget class="QWidget" name="tabIqtFit">
+      <widget class="QWidget" name="tabIqt">
        <attribute name="title">
-        <string>I(Q, t) Fit</string>
+        <string>I(Q, t) </string>
```
Comment (robertapplin) [inline qt/scientific_interfaces/Inelastic/QENSFitting/QENSFitting.ui:72]:
```suggestion
        <string>I(Q, t)</string>
```

**Review by robertapplin — APPROVED** [review]
Thanks for making those changes! I've built the dev-docs and normal docs, and they build fine. All the references to the tabs use the new names, and I've followed the manual testing instructions to make sure QENS Fitting is still working. I can also confirm the Settings interface is still working 👍

**Pull request merged**